apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/sync/SyncDocumentWrapper.kt

```diff
@@ -0,0 +1,67 @@
+package com.twilio.livevideo.app.manager.sync
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.twilio.sync.ErrorInfo
+import com.twilio.sync.EventContext
+import com.twilio.sync.SuccessListener
+import com.twilio.sync.SyncClient
+import com.twilio.sync.SyncDocument
+import com.twilio.sync.SyncDocumentObserver
+import com.twilio.sync.SyncOptions
+import org.json.JSONObject
+import timber.log.Timber
+
+class SyncDocumentWrapper constructor(var uniqueName: String = "") : SyncObject {
+
+    private var document: SyncDocument? = null
+
+    private val _onStateEvent: MutableLiveData<SyncViewEvent?> =
+        MutableLiveData<SyncViewEvent?>(null)
+    val onStateEvent: LiveData<SyncViewEvent?>
+        get() {
+            val event = _onStateEvent
+            _onStateEvent.value = null
+            return event
+        }
+
+    private val syncDocumentObserver: SyncDocumentObserver = object : SyncDocumentObserver() {
+        override fun onUpdated(context: EventContext?, data: JSONObject?, previousData: JSONObject?) {
+            super.onUpdated(context, data, previousData)
+
+            data?.apply {
+                val inviteKey = "speaker_invite"
```

Comment:
This code makes this object specific to the `user` sync document. And so I think it would be good to have `user` somewhere in the name of the class. Otherwise the name of this class implies it is for any kind of sync document.

Reply:
Done!